pkg/kf/commands/quotas/integration_test.go

```diff
@@ -0,0 +1,184 @@
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+		routesQuantity := "100"
+
+		kf.CreateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+			"-r", routesQuantity,
+		)
+
+		getQuotaOutput := kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, " "), []string{quotaName, memQuantity, cpuQuantity, routesQuantity})
+
+		deleteQuotaOutput := kf.DeleteQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(deleteQuotaOutput, " "), []string{quotaName, "successfully deleted"})
+
+		// Getting the quota should output an error saying the quota is not found
+		getQuotaOutput = kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, " "), []string{quotaName, "Error", "not found"})
+	})
+}
+
+// TestIntegration_Update creates a quota then updates it.
+// It checks that the quota is saved with the new values.
+// Afterwards, the quota is deleted.
+func TestIntegration_Update(t *testing.T) {
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+		defer kf.DeleteQuota(ctx, quotaName)
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+
+		kf.CreateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+		)
+		// Check that routes quota is defaulted to "0"
+		getQuotaOutput := kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, "\n"), []string{quotaName, memQuantity, cpuQuantity, "0"})
+		Logf(t, "done ensuring correct quota info returned.")
+
+		memQuantity = "600Gi"
+		cpuQuantity = "12"
+		routesQuantity := "100"
+
+		kf.UpdateQuota(ctx, quotaName,
+			"-m", memQuantity,
+			"-c", cpuQuantity,
+			"-r", routesQuantity,
+		)
+
+		getQuotaOutput = kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, "\n"), []string{quotaName, memQuantity, cpuQuantity, routesQuantity})
+		Logf(t, "done ensuring quota info updated.")
+
+		// Reset memory quota
+		memQuantity = "0"
+		kf.UpdateQuota(ctx, quotaName,
+			"-m", memQuantity,
+		)
+
+		getQuotaOutput = kf.GetQuota(ctx, quotaName)
+		AssertContainsAll(t, strings.Join(getQuotaOutput, "\n"), []string{quotaName, memQuantity, cpuQuantity, routesQuantity})
+		Logf(t, "done ensuring memory quota was reset.")
+	})
+}
+
+// TestIntegration_List creates multiple quotas, then checks that all of them are listed.
+// Afterwards, all of the quotas are deleted.
+func TestIntegration_List(t *testing.T) {
+	checkClusterStatus(t)
+	RunKfTest(t, func(ctx context.Context, t *testing.T, kf *Kf) {
+		quotaName := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+		quota2Name := fmt.Sprintf("integration-quota-%d", time.Now().UnixNano())
+		defer kf.DeleteQuota(ctx, quotaName)
+		defer kf.DeleteQuota(ctx, quota2Name)
+
+		memQuantity := "500Gi"
+		cpuQuantity := "10"
+		routesQuantity := "100"
```

Comment:
I think routesQuantity should have a different leading digit so when the assertcontainsall looks for one it doesn't find the other and think the condition is met if there's an accidental change

Reply:
updated all the values in the tests to have different digits to avoid this issue